### PR TITLE
Implement Log Storage Bucket

### PR DESCRIPTION
**Pull request by osterman**
## what
* Create an S3 bucket suitable for receiving logs from other AWS services
* Implement a configurable log retention policy

## why
* S3, CloudFront, and CloudTrails generate an enormous amount of logs
* Expunging old logs is the only scalable solution

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+| `lifecycle_rule_enabled`            | `true`              | Enable object lifecycle rules on this bucket                                            | No       |
+| `versioning_enabled`                | `false`             | Versioning is a means of keeping multiple variants of an object in the same bucket      | No       |
+| `noncurrent_version_transition_days`| `30`                | Number of days to persist in the standard storage tier before moving to the glacier tier| No       |
+| `noncurrent_version_expiration_days`| `90`                | Specifies when noncurrent object versions expire                                        | No       |
+| `standard_transition_days`          | `30`                | Number of days to persist in the standard storage tier before moving to the glacier tier| No       |
+| `glacier_transition_days`           | `60`                | Number of days after which to move the data to the glacier storage tier                 | No       |
+| `glacier_transition_days`           | `90`                | Number of days after which to expunge the objects                                       | No       |
```
Comment (comeanother) [inline README.md]: Rename to `expiration_days`.

**Comment by comeanother** [comment]
Tested.
screenshot from 2017-09-15 18-19-14

File: README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+| `region`                            | ``                  | If specified, the AWS region this bucket should reside in. Defaults to region of callee.| No       |
+| `force_destroy`                     | ``                  | All objects will be forcefully deleted from the bucket when bucket destroyed            | No       |
+| `lifecycle_rule_enabled`            | `true`              | Enable object lifecycle rules on this bucket                                            | No       |
+| `versioning_enabled`                | `false`             | Versioning is a means of keeping multiple variants of an object in the same bucket      | No       |
+| `noncurrent_version_transition_days`| `30`                | Number of days to persist in the standard storage tier before moving to the glacier tier| No       |
+| `noncurrent_version_expiration_days`| `90`                | Specifies when noncurrent object versions expire                                        | No       |
+| `standard_transition_days`          | `30`                | Number of days to persist in the standard storage tier before moving to the glacier tier| No       |
```
Comment (darend) [inline README.md]: @osterman this should be "Number of days to persist in the standard storage tier before moving to the ~glacier tier~ infrequent access tier"

File: variables.tf [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  default     = "90"
+}
+
+variable "noncurrent_version_transition_days" {
+  description = "(Optional) Specifies when noncurrent object versions transitions"
+  default     = "30"
+}
+
+variable "standard_transition_days" {
+  description = "Number of days to persist in the standard storage tier before moving to the glacier tier"
```
Comment (darend) [inline variables.tf]: ~glacier~ infrequent access

**Pull request merged**